docs/dev_setup.rst

```diff
@@ -11,12 +11,58 @@ This section describes how someone wanting to contribute to the libCellML projec
+This document provides a brief overview on how to satisfy these pre-requisites but it doesn't go into great detail (Google is your friend) as there are many great resources for acquiring and installing the required software.  In this document we will detail the software that we currently use but please be aware that software is always being updated and the documentation may not reflect the latest available.
+
+Pre-requisite acquisition
+=========================
+
+In this section we cover the retrieval and installation of pre-requisites.
+
+Git
+---
+
+Creating a GitHub user account is straightforward and can be done `here <https://github.com/join>`_.  Installing a *git* client is particular to each operating system and some pointers are offered below:
+
+* macOS *git* is pre-installed and available from the command line.
+* Ubuntu (and other linux distributions) *git* can be installed using the package manager with the command :code:`sudo apt-get install git`.
+* Windows *git* is available from a variety of vendors we commonly use `git for windows <http://gitforwindows.org/>`_, but other popular *git* implementations are:
+
+  - `GitHub Desktop <https://desktop.github.com/>`_
+  - `GitKracken <https://www.gitkraken.com/>`_
+  - `Git SCM <https://git-scm.com/>`_
+
+  The choice of implementation is one of personal preference.
+
+CMake
+-----
+
+*CMake* is the cross-platform family of tools designed to build, test and package software.  *CMake* is used to control the software compilation process using simple platform and compiler independent configuration files, and generate native makefiles and workspaces that can be used in the compiler environment of your choice.
+
+Again installation of *CMake* is particular to each operating system.  For Ubuntu (and other linux distributions) *CMake* can be installed using the package manager with the command :code:`sudo apt-get install cmake`.  For macOS and Windows *CMake* provides installation binaries from `here <https://cmake.org/download/>`_ choose the binary appropriate for your operating system and follow the installation instructions.
+
+Toolchain
+---------
+
+The toolchain specifies the compiler that we will use to build *libCellML*, toolchains are highly dependent on the operating system.  When we test *libCellML* we currenly use the *clang* family of compilers on macOS, the *GNU* family of compilers on Ubuntu, and *Microsoft Visual Studio Compilers* on Windows.  We recommend using these compilers on these systems, but feel free to use a different toolchain.  We often use *intel* compilers to build *libCellML* but we don't (at the time of writing) test with them.
+
+The following sub-sections below provide guidance on how to install the recommended toolchain on the major operating systems that *libCellML* supports.
```

Comment:
The following sub-sections provide guidance on how to install the recommended toolchain on the major operating systems that *libCellML* supports.